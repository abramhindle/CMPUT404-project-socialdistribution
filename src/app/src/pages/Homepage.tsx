import React from 'react';
import styled from 'styled-components';
import Logo from '../components/Logo';

// const NavContainer = styled.div`
//   width: 100%;
//   top: 0%;
//   position: fixed;
//   text-align: center;
//   height: 5%;
//   border-bottom: 1px solid;
// `;

const HomepageContainer = styled.div`
  width: 100%;
  flex: display;
  align: center;
  height: 100%;
  background-color: orange;
`;
export default function Homepage() {
  return (
    <HomepageContainer>
      <Logo />
    </HomepageContainer>
  );
}
