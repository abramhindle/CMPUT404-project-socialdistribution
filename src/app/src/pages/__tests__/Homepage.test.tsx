import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import Homepage from '../Homepage';

let container: any = null;
//issues with container type and its methods

beforeEach(() => {
  // setup a DOM element as a render target
  container = document?.createElement('div');
  document?.body?.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container?.remove();
  container = null;
});

describe('Homepage suite', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<Homepage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
