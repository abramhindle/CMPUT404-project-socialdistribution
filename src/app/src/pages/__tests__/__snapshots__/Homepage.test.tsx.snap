// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage suite Renders correctly 1`] = `
<div
  className="sc-gsDKAQ kFsreC"
>
  <div
    className="sc-dkPtRN fJxRa-D"
  >
    <div
      className="sc-bdvvtL bxQDqD"
    >
      Website.
    </div>
  </div>
  <div
    className="sc-hKwDye hFjOEP"
  >
    Welcome to a Modern Social Media Website.
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1pguk2u-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Lets get started
    </button>
  </div>
</div>
`;
